Source/Fuse.Elements/Resources/Exif/Exif.uno

```diff
@@ -0,0 +1,242 @@
+			{
+				if (p.Id == Exif.OrientationTagId)
+				{
+					var b = new Buffer(p.Value);
+					var orientation = b.GetShort(0,true);
+					return orientation;
+				}
+			}
+			return 0;
+		}
+	}
+
+	[Require("Source.Include", "ImageIO/ImageIO.h")]
+	[Require("Source.Include", "CoreFoundation/CoreFoundation.h")]
+	[Require("Source.Include", "ExifHelper.h")]
+	[Require("Xcode.Framework", "ImageIO.framework")]
+	[Require("Xcode.Framework", "CoreFoundation.framework")]
+	extern(iOS) class ExifIOSImpl : IExifImpl
+	{
+		byte[] _bytes; 
+
+		public ExifIOSImpl(byte[] bytes)
+		{
+			_bytes = bytes;
+		}
+
+		byte ReadByte(int pos) {
+			return _bytes[pos];
+		}
+
+		public int GetOrientation()
+		{
+			return GetOrientationImpl(_bytes.Length, ReadByte);
+		}
+
+		[Foreign(Language.ObjC)]
+		int GetOrientationImpl(int length, Func<int,byte> readByte)
+		@{
+
+			CGDataProviderDirectCallbacks callbacks = {
+				.getBytePointer = 0,
+				.getBytesAtPosition = getBytesAtPosition
+			};
+			UnoCallbackBox box = { readByte };
+
+			CGDataProviderRef dataProvider = CGDataProviderCreateDirect(&box, length, &callbacks);
+			CGImageSourceRef source = CGImageSourceCreateWithDataProvider(dataProvider, NULL);
+			NSArray *metadataArray = nil;
+
+			if (dataProvider) {
+				if (source) {
+					CGImageMetadataRef metadata = CGImageSourceCopyMetadataAtIndex(source, 0, NULL);
+					if (metadata) {
+						metadataArray = CFBridgingRelease(CGImageMetadataCopyTags(metadata));
+						CFRelease(metadata);
+					}
+					CFRelease(source);
+				}
+				CGDataProviderRelease(dataProvider);
+			}
+
+			int rotation = 0;
+			NSString* tagTarget = @"Orientation";
+			for (id tag in metadataArray) {
+				CFStringRef tagName = CGImageMetadataTagCopyName((CGImageMetadataTagRef)tag);
+				if ([tagTarget isEqualToString: (__bridge NSString*)tagName]) {
+					CFTypeRef rot = CGImageMetadataTagCopyValue((CGImageMetadataTagRef)tag);
+					rotation = [((__bridge NSNumber*)rot) intValue];
+				}
+			}
+
+			return rotation;
+
+		@}
+	}
+
+	[Require("Gradle.Dependency.Compile", "com.drewnoakes:metadata-extractor:2.10.1")]
+	[ForeignInclude(Language.Java,
+		"java.io.ByteArrayInputStream",
+		"java.io.InputStream",
+		"java.io.IOException",
+		"com.drew.imaging.ImageMetadataReader",
+		"com.drew.imaging.ImageProcessingException",
+		"com.drew.metadata.Metadata",
+		"com.drew.metadata.MetadataException",
+		"com.drew.metadata.exif.ExifIFD0Directory")]
+	//Uses https://drewnoakes.com/code/exif/
+	extern(Android) class ExifAndroidImpl : IExifImpl
+	{
+		byte[] _buffer;
+		public ExifAndroidImpl(byte[] buffer)
+		{
+			_buffer = buffer;
+		}
+
+		public int GetOrientation()
+		{
+			var stream = CreateStream(ReadByte);
+			return GetOrientationImpl(stream);
+		}
+
+		int _position = 0;
+		int ReadByte()
+		{
+			if (_position >= _buffer.Length)
+			{
+				return -1;
+			}
+			return (int)_buffer[_position++];
+		}
+
+		[Foreign(Language.Java)]
+		Java.Object CreateStream(Func<int> readByte)
+		@{
```

Comment:
Similarly to on iOS, you shouldn't need this. You can create an InputStream from a byte-buffer using this helper:
https://github.com/fusetools/fuselibs-public/blob/master/Source/Fuse.Android/AndroidInterop.uno#L66